src/hardwareio/binaryDumper/binary-dumper.cpp

```diff
@@ -0,0 +1,236 @@
+        }
+
+        info_dumper.close();
+        event_dumper.close();
+    }
+
+    //synchronous thread
+    virtual bool updateModule()
+    {
+
+        //add any synchronous operations here, visualisation, debug out prints
+        std::cout << "\r" << packets_saved << " packets saved. "
+                  << input_port.queryunprocessed() << " packets in queue."
+                  << "                                                    ";
+        std::cout.flush();
+
+        return Thread::isRunning();
+    }
+
+    //asynchronous thread run forever
+    void run()
+    {
+        Stamp yarpstamp;
+        total_zynq_dt = 0;
+
+        while(true) {
+
+            //get the first packet (blocking read)
+            const vector<int32_t> * q = input_port.read(yarpstamp);
+            if(!q || Thread::isStopping()) return;
+
+            //only when the first packet is read do our first initialisation
+            if(save_first) {
+                //save first stamps (zynq and local cpu)
+                first_zynq_time = yarpstamp.getTime();
+                first_cpu_time = yarp::os::Time::now();
+
+                //do a small correction so we are aligned to the first event
+                //in the packet, and not the last
+                double correction_factor = (*q)[q->size() - 2] - (*q)[0];
+                if(correction_factor < 0) correction_factor += vtsHelper::max_stamp;
+                correction_factor *= vtsHelper::tsscaler;
+                yInfo() << "Correction factor" << correction_factor;
+                first_zynq_time -= correction_factor;
+                first_cpu_time -= correction_factor;
+
+                //dump our corrected cpu time
+                info_dumper << "[" << first_cpu_time  << "] "
+                            << port_name_reader.incoming_port_name
+                            << " [connected]" << std::endl;
+                save_first = false;
+            }
+
+            //write the event-stream to binary file
+            event_dumper.write((const char *)q->data(), q->size() * sizeof(int32_t));
+
+            //update the total time of the data-stream
+            //from zynq clock (irrespective of any "back-load" of packets to save)
+            total_zynq_dt = yarpstamp.getTime() - first_zynq_time;
+            last_cpu_time = Time::now();
+
+            //count our progress
+            packets_saved++;
+
+            //feed the data through if needed
+            if(output_port.getOutputCount())
+                output_port.write(*q, yarpstamp);
+        }
+    }
+};
+
+
+
+int main(int argc, char * argv[])
+{
+    /* initialize yarp network */
+    yarp::os::Network yarp;
+    if(!yarp.checkNetwork(2)) {
+        std::cout << "Could not connect to YARP" << std::endl;
+        return false;
+    }
+
+    /* prepare and configure the resource finder */
+    yarp::os::ResourceFinder rf;
+    rf.setVerbose( false );
+    rf.setDefaultContext( "eventdriven" );
```

Comment:
```suggestion
    rf.setDefaultContext( "event-driven" );
```

Reply:
this change is for the binary-dumper and therefore shouldn't be in this merge request at all. I've fixed that - but your actual comment is correct. Could you go through the entire event-driven library and fix this same issue on all the modules? thanks? then open a pull request :)